runtime/src/main/cpp/Memory.cpp

```diff
@@ -1997,6 +1997,16 @@ inline void checkIfGcNeeded(MemoryState* state) {
   }
 }
 
+inline void checkIfForceCyclicGcNeeded(MemoryState* state) {
+  if (state != nullptr && state->toFree->size() > kMaxToFreeSizeThreshold) {
+    // To avoid GC trashing check that at least 10ms passed since last GC.
+    if (konan::getTimeMicros() - state->lastGcTimestamp > 10 * 1000) {
+      GC_LOG("Calling GC from checkIfForceCyclicGcNeeded: %d\n", state->toFree->size())
+      garbageCollect(state, true);
```

Comment:
I guess forced cycle collection shouldn't trigger increasing collect cycles threshold in `garbageCollect`.

Reply:
Only collect cycles threshold ?
I now think about same logic for gcThreshold.

Reply:
Well, makes sense.

Reply:
I made such logic only for increasing based on time.